app/src/androidTest/kotlin/com/k0d4black/theforce/TheForceTestApplication.kt

```diff
@@ -1,18 +1,29 @@
 package com.k0d4black.theforce
 
-import com.k0d4black.theforce.di.AppComponent
-import com.k0d4black.theforce.di.DaggerTestApplicationComponent
+import org.koin.android.ext.koin.androidContext
+import org.koin.android.ext.koin.androidLogger
+import org.koin.core.context.loadKoinModules
+import org.koin.core.context.startKoin
+import org.koin.core.context.unloadKoinModules
+import org.koin.core.module.Module
 
 class TheForceTestApplication : TheForceApplication() {
 
     override fun onCreate() {
         super.onCreate()
-        getApplicationComponent().inject(this)
+        startKoin {
+            androidLogger()
+            androidContext(this@TheForceTestApplication)
+            startKoin {
+                modules(emptyList())
+            }
+        }
```

Comment:
- Next, the ```startKoin``` DSL block has a nested startKoin block which will re-introduce the error mentioned above. Remove it, leaving this as the result.
```kotlin
startKoin{
            androidLogger()
            androidContext(this@TheForceTestApplication)
            modules(emptyList())
     }
```
- Next, since the test application is running independently of the main application it needs the modules declared just as the ones in the ```ForceTestApplication``` only with a slight modification on the ```networkingModule``` using the ```fakeNetworkingModule```.The result of all that is
```kotlin
class TheForceTestApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        startKoin {
            androidLogger()
            androidContext(this@TheForceTestApplication)
            modules(
                fakeNetworkModule,
                viewModelsModule,
                repositoriesModule,
                dataSourceModule,
                useCasesModule
            )
        }
    }

}
```
Alternatively, you could also look into overriding modules for testing but this solves it for now.

Reply:
opps, rigth I was actually very close to pass all test cases 😄 👊